### PR TITLE
chore: Enable -Werror=unused-parameter flag

### LINKED ISSUE #2266: Treat `unused-parameter` warnings as errors in CMake
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

See https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Warning-Options.html.

### Solution

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by c8ef**
- resolve: #2266
- supersede: #2313

### CODE REVIEW

File: src/vendor/rand.cc [hunk 1]
```diff
@@ -106,11 +106,6 @@ constexpr void SEED(const T x0, const T x1, const T x2) {
   SET3(x, x0, x1, x2), SET3(a, A0, A1, A2), c = C;
 }
 
-template <typename T>
-constexpr T HiBit(const T x) {
```
Comment (c8ef) [inline src/vendor/rand.cc:110]: This function appears to be unused.

Comment (mapleFU) [reply src/vendor/rand.cc:110]: this is vendored ( pasted from other project )

Maybe we can disable lint for "vendor" dir?

Comment (c8ef) [reply src/vendor/rand.cc:110]: Even though it's in the vendored directory, I believe it's more of a modern C++ re-implementation based on the comment below. Therefore, removing the unused function should be acceptable, at least NFC.

https://github.com/apache/kvrocks/blob/5ecf023bcf0a92e23982fef1710a53f495cfef9a/src/vendor/rand.cc#L44-L47

**Comment by c8ef** [comment]
I don't know why [this test](https://github.com/apache/kvrocks/actions/runs/10562102251/job/29261542437#logs) failed. I believe the entire patch should be NFC. Other CI failures have already been fixed in the latest patch. Please allow the CI to run.

**Review by git-hulk — APPROVED** [review]

**Review by torwig — APPROVED** [review]

**Pull request merged**